plugin/core/state.py

```
@@ -26,19 +66,374 @@ def __init__(self):
+        if not isinstance(message_id, str) or not message_id:
+            return False
+        with self._bus_store_lock:
+            self._deleted_message_ids.add(message_id)
+            removed = False
+            for idx, rec in enumerate(list(self._message_store)):
+                if isinstance(rec, dict) and rec.get("message_id") == message_id:
+                    try:
+                        del self._message_store[idx]
+                        removed = True
+                        break
+                    except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)